### PR TITLE
Change where() statements to clip()

### LINKED ISSUE #60: replace where with clip in eos
Here:
https://github.com/NCAR/pop-tools/blob/b83e905f99d89ffcfac3ca513ebac55455b648ac/pop_tools/eos.py#L96-L100

and here:
https://github.com/NCAR/pop-tools/blob/b83e905f99d89ffcfac3ca513ebac55455b648ac/pop_tools/eos.py#L134-L137

This should be ≈ 2x speedup
image

### PR DESCRIPTION
**Pull request by mnlevy1981**
Fixes #60 

I was hoping to be able to put

```
temp = temp.clip(tmin, tmax)
salt = salt.clip(smin, smax)
```

outside the if statement, but ran into

```
==================================================== short test summary info =====================================================
FAILED tests/test_eos.py::test_eos_numpy_1 - AttributeError: 'float' object has no attribute 'clip'
FAILED tests/test_eos.py::test_eos_numpy_2 - AttributeError: 'float' object has no attribute 'clip'
FAILED tests/test_eos.py::test_eos_numpy_3 - AttributeError: 'float' object has no attribute 'clip'

```

in testing

### CODE REVIEW

File: pop_tools/eos.py [hunk 1]
```diff
@@ -94,10 +94,8 @@ def eos(salt, temp, return_coefs=False, **kwargs):
     smax = 999.0
 
     if use_xarray:
-        temp = xr.where(temp < tmin, tmin, temp)
-        temp = xr.where(temp > tmax, tmax, temp)
-        salt = xr.where(salt < smin, smin, salt)
-        salt = xr.where(salt > smax, smax, salt)
+        temp = temp.clip(tmin, tmax)
+        salt = salt.clip(smin, smax)
```
Comment (andersy005) [inline pop_tools/eos.py:98]: It's my understanding that 

Line 97-98

```python
temp = temp.clip(tmin, tmax)
salt = salt.clip(smin, smax)
```

and 

Line 134-135

```python
temp = np.clip(temp, tmin, tmax)
salt = np.clip(salt, smin, smax)
```

are doing exactly the same thing, and can be consolidated into just 

```python
temp = np.clip(temp, tmin, tmax)
salt = np.clip(salt, smin, smax)
```

which will return an xarray object or a numpy array depending on the type of the input.

Comment (andersy005) [reply pop_tools/eos.py:98]: Never mind ;) https://github.com/NCAR/pop-tools/pull/63#issue-478211186

Comment (dcherian) [reply pop_tools/eos.py:98]: There is one unfortunate difference: https://github.com/pydata/xarray/issues/2570

Can we put a min numpy of 1.17? I'm not sure it affects anyone really and saves some effort.  1.17.0 was released in June 2019.

**Review by dcherian — APPROVED** [review]
Thanks @mnlevy1981

**Review by andersy005 — APPROVED** [review]
Looks good to me 👍

**Pull request merged**